README.md

```diff
@@ -1040,4 +1040,514 @@ Other Style Guides
+
+    ```javascript
+    // we know this wouldn't work (assuming there
+    // is no notDefined global variable)
+    function example() {
+        console.log(notDefined); // => throws a ReferenceError
+    }
+
+    // creating a variable declaration after you
+    // reference the variable will work due to
+    // variable hoisting. Note: the assignment
+    // value of `true` is not hoisted.
+    function example() {
+        console.log(declaredButNotAssigned); // => undefined
+        var declaredButNotAssigned = true;
+    }
+
+    // The interpreter is hoisting the variable
+    // declaration to the top of the scope,
+    // which means our example could be rewritten as:
+    function example() {
+        let declaredButNotAssigned;
+        console.log(declaredButNotAssigned); // => undefined
+        declaredButNotAssigned = true;
+    }
+
+    // using const and let
+    function example() {
+        console.log(declaredButNotAssigned); // => throws a ReferenceError
+        console.log(typeof declaredButNotAssigned); // => throws a ReferenceError
+        const declaredButNotAssigned = true;
+    }
+    ```
+
+  - [14.2](#14.2) <a name='14.2'></a> Anonymous function expressions hoist their variable name, but not the function assignment.
+
+    ```javascript
+    function example() {
+        console.log(anonymous); // => undefined
+
+        anonymous(); // => TypeError anonymous is not a function
+
+        var anonymous = function() {
+            console.log('anonymous function expression');
+        };
+    }
+    ```
+
+  - [14.3](#14.3) <a name='14.3'></a> Named function expressions hoist the variable name, not the function name or the function body.
+
+    ```javascript
+    function example() {
+        console.log(named); // => undefined
+
+        named(); // => TypeError named is not a function
+
+        superPower(); // => ReferenceError superPower is not defined
+
+        var named = function superPower() {
+            console.log('Flying');
+        };
+    }
+
+    // the same is true when the function name
+    // is the same as the variable name.
+    function example() {
+        console.log(named); // => undefined
+
+        named(); // => TypeError named is not a function
+
+        var named = function named() {
+            console.log('named');
+        }
+    }
+    ```
+
+  - [14.4](#14.4) <a name='14.4'></a> Function declarations hoist their name and the function body.
+
+    ```javascript
+    function example() {
+        superPower(); // => Flying
+
+        function superPower() {
+            console.log('Flying');
+        }
+    }
+    ```
+
+  - [14.5](#14.5) <a name='14.5'></a> ES6 `import`s are hoisted to the beginning of their module.
+
+    ```javascript
+    // This works
+    foo();
+
+    import { foo } from 'my_module';
+
+    // This will import 'imported_module' before 'required_module'
```

Comment:
Is it just me or doesn't this comment make any sense?

I looked this section up in airbnb/javascript and they removed it: https://github.com/airbnb/javascript/blob/master/README.md#14.4

What it's meant to be saying is:

ES6 imports are hoisted to the top of the file.
Using the `require` syntax to import modules does not apply to this.
Therefore, the example should be:

```javascript
// This works
foo();

import { foo } from 'my_module';

// This will import 'my_module' before 'required_module'
require('required_module');

import 'my_module';
```

Correct me, if I'm wrong.